src/MVVM/Factories/ViewModelFactory.cs

```diff
@@ -88,18 +87,102 @@ internal static INavigationService TryResolveNavigationServiceInstance()
+
+                if (initialize)
+                    TaskHelper.RunSync(instance.InitializeAsync);
+
+                return instance;
+            }
+            catch (Exception innerException)
+            {
+                string errorMessage = $"ViewModel of type {interfaceType.Name} cannot be resolved --- is it registered?";
+
+                log?.Critical(
+                    errorMessage,
+                    innerException);
+
+                throw new TypeLoadException(
+                    errorMessage,
+                    innerException);
+            }
+        }
+
+
+
+        /// <summary>
+        /// Creates a new ViewModel instance and optionally invokes the <see cref="IViewModel.InitializeAsync"/> method.
+        /// </summary>
+        /// <typeparam name="TViewModel">Type of the ViewModel to create</typeparam>
+        /// <param name="initialize">Specifies if <see cref="IViewModel.InitializeAsync"/> should get called after instance creation.</param>
+        /// <returns>A ViewModel instance of the given type</returns>
+        public static async Task<TViewModel> ResolveAsync<TViewModel>(
+            bool initialize = true)
+            where TViewModel : class, IViewModel
+        {
+            try
+            {
+                var instance = iocContainer.Resolve<TViewModel>();
+
+                if (initialize)
+                    await instance.InitializeAsync();
+
+                return instance;
+            }
+            catch (Exception innerException)
+            {
+                string errorMessage = $"ViewModel of type {typeof(TViewModel).Name} cannot be resolved --- is it registered?";
+
+                log?.Critical(
+                    errorMessage,
+                    innerException);
+
+                throw new TypeLoadException(
+                    errorMessage,
+                    innerException);
+            }
+        }
+
+        /// <summary>
+        /// Creates a new ViewModel instance and optionally invokes the <see cref="IViewModel.InitializeAsync"/> method.
+        /// </summary>
+        /// <param name="viewModelType">Type of the ViewModel to create</param>
+        /// <param name="initialize">Specifies if <see cref="IViewModel.InitializeAsync"/> should get called after instance creation.</param>
+        /// <returns>A ViewModel instance of the given type</returns>
+        public static async Task<IViewModel> ResolveAsync(
+            Type viewModelType,
+            bool initialize = true)
+        {
+            var interfaceType = typeof(IViewModel);
+
             try
             {
-                var instance = iocContainer.Resolve<TViewModelInterface>();
+                if (!interfaceType.IsAssignableFrom(viewModelType))
+                {
+                    throw new InvalidOperationException(
+                        $"The type {viewModelType.Name} must implement interface type {interfaceType.Name}");
```

Comment:
Codacy found an issue: [Refactor this method to add validation of parameter 'viewModelType' before using it.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5767060)